--- a/tools/clang/scripts/update.py
+++ b/tools/clang/scripts/update.py
@@ -358,6 +358,13 @@
       print(force_head_revision)
       return 0
 
+    stamp_version = ReadStampFile(STAMP_FILE).partition(',')[0]
+    if PACKAGE_VERSION != stamp_version:
+      print('The expected clang version is %s but the actual version is %s' %
+            (PACKAGE_VERSION, stamp_version))
+      print('Did you run "gclient sync"?')
+      return 1
+
     print(PACKAGE_VERSION)
     return 0
 
