--- ./tools/grit/grit/format/gzip_string.py
+++ ./tools/grit/grit/format/gzip_string.py
@@ -18,8 +18,10 @@
   # Instead, --rsyncable breaks the file into small chunks, so that one doesn't
   # affect the other in compression, and then only that chunk will have to be
   # updated.
-  gzip_proc = subprocess.Popen(['gzip', '--stdout', '--rsyncable',
-                                '--best', '--no-name'],
+  #
+  # busybox gzip does not support --rsyncable.
+  gzip_proc = subprocess.Popen(['gzip', '-c',
+                                '-9'],
                                stdin=subprocess.PIPE,
                                stdout=subprocess.PIPE,
                                stderr=subprocess.PIPE)
