--- ./tools/gn/bootstrap/bootstrap.py
+++ ./tools/gn/bootstrap/bootstrap.py
@@ -121,7 +121,7 @@
   shutil.copy2(
       os.path.join(BOOTSTRAP_DIR, 'last_commit_position.h'), gn_build_dir)
   subprocess.check_call(
-      [ninja_binary, '-C', gn_build_dir, '-w', 'dupbuild=err', 'gn'])
+      [ninja_binary, '-C', gn_build_dir, '-j'+str(options.jobs), 'gn'])
   shutil.copy2(os.path.join(gn_build_dir, 'gn'), gn_path)
 
   if not options.skip_generate_buildfiles:
