--- ./base/allocator/allocator.gni
+++ ./base/allocator/allocator.gni
@@ -27,10 +27,10 @@
 
 declare_args() {
   # Memory allocator to use. Set to "none" to use default allocator.
-  use_allocator = _default_allocator
+  use_allocator = "none"
 
   # Causes all the allocations to be routed via allocator_shim.cc.
-  use_allocator_shim = _default_use_allocator_shim
+  use_allocator_shim = false
 }
 
 assert(use_allocator == "none" || use_allocator == "tcmalloc" ||
